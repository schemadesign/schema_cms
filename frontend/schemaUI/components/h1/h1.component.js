import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import { headerStyle } from './h1.styles';

export class H1 extends PureComponent {
  static propTypes = {
    children: PropTypes.oneOfType([PropTypes.element, PropTypes.string]),
    style: PropTypes.object,
  };

  render() {
    const {style: componentStyle = {}, children} = this.props;
    const style = {...headerStyle, ...componentStyle};

    return <h1 style={style} >{children}</h1>;
  }
}
