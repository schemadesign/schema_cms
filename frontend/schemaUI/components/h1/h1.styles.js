export const headerStyles = {
  fontSize: '24px',
};
