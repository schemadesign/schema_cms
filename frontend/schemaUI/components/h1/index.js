export { H1 } from './h1.component';
