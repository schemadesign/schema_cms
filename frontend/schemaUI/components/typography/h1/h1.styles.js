import { primary } from '../../../utils/theme';

export const headerStyles = {
  fontSize: '24px',
  fontFamily: primary.fontFamily,
  fontWeight: 600,
};
