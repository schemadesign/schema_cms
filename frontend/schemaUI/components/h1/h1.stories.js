import React from 'react';
import { storiesOf } from '@storybook/react';

import { H1 } from './h1.component';

storiesOf('H1', module).add('Default', () => (
  <H1>
    Some title
  </H1>
));

const defaultProps = {
  style: {
    color: 'red',
    'text-decoration': 'underline',
  },
};

storiesOf('H1', module).add('with style', () => (
  <H1 {...defaultProps} >
    Some title
  </H1>
));