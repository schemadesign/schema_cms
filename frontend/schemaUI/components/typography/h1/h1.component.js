import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import { primary } from '../../../utils/theme';
import { headerStyles } from './h1.styles';

export class H1 extends PureComponent {
  static propTypes = {
    children: PropTypes.oneOfType([PropTypes.element, PropTypes.string]),
    customStyles: PropTypes.object,
    theme: PropTypes.object,
  };

  render() {
    const { customStyles = {}, theme = primary, children, ...restProps } = this.props;
    const styles = { ...headerStyles, ...theme.typography.h1, ...customStyles };

    return (
      <h1 style={styles} {...restProps}>
        {children}
      </h1>
    );
  }
}
