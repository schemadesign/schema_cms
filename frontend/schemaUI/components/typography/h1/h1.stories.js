import React from 'react';
import { storiesOf } from '@storybook/react';

import { H1 } from './h1.component';

storiesOf('Typography.H1', module).add('Default', () => <H1>Some title</H1>);

const withStylesProps = {
  customStyles: {
    color: 'red',
    textDecoration: 'underline',
    fontStyle: 'italic',
    fontSize: '30px',
  },
};

storiesOf('Typography/H1', module).add('with styles', () => <H1 {...withStylesProps}>Some title</H1>);

storiesOf('Typography/H1', module).add('with attributes', () => (
  <H1 title="Oh! It's title." id="title-with-attribute">
    Title with attributes
    <span style={{ fontSize: '10px', color: '#777', display: 'block' }}>
      (hover on it to show value of title attribute)
    </span>
  </H1>
));

storiesOf('Typography/H1', module).add('with custom theme', () => {
  const theme = {
    typography: {
      h1: {
        fontFamily: 'serif',
        size: '30px',
        color: 'orange',
        border: '1px solid gray',
        padding: '10px',
        display: 'inline-block',
      },
    },
  };

  return <H1 theme={theme}>Title styled by theme</H1>;
});
