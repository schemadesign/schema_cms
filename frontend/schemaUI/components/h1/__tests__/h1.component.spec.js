import React from 'react';
import { shallow } from 'enzyme';

import { H1 } from '../h1.component';

describe('H1: Component', () => {
  const defaultProps = {};

  const component = props => <H1 {...defaultProps} {...props} />;

  const render = (props = {}) => shallow(component(props));

  it('should render correctly', () => {
    const wrapper = render();
    global.expect(wrapper).toMatchSnapshot();
  });

  it('should render children when passed in', () => {
    const wrapper = shallow((
      <H1>
        Super title
      </H1>
    ));

    global.expect(wrapper).toMatchSnapshot();
  });

  it('should render with custom styles', () => {
    const customStyles = {color: 'blue'};

    const wrapper = shallow((
      <H1 customStyles={customStyles}>
        Blue title
      </H1>
    ));

    global.expect(wrapper).toMatchSnapshot();
  });
});
