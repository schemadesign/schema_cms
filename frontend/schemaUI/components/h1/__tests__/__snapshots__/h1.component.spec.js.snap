// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`H1: Component should render children when passed in 1`] = `
<h1
  style={
    Object {
      "fontSize": "24px",
    }
  }
>
  Super title
</h1>
`;

exports[`H1: Component should render correctly 1`] = `
<h1
  style={
    Object {
      "fontSize": "24px",
    }
  }
/>
`;

exports[`H1: Component should render with custom styles 1`] = `
<h1
  style={
    Object {
      "color": "blue",
      "fontSize": "24px",
    }
  }
>
  Blue title
</h1>
`;
